src/main/python/community-detector/community_detector.py

```diff
@@ -0,0 +1,141 @@
+from collections import defaultdict
+from itertools import chain
+import logging
+from igraph import Graph
+import numpy
+
+
+def detect_communities(cc,
+                       buckets_matrix,
+                       edges="linear",
+                       algorithm="walktrap",
+                       algorithm_params={}):
+    """
+    Runs Community Detection analysis on the given connected components.
+    :param cc: numpy.ndarray with the connected components
+    :param buckets_matrix: scipy.sparse.csr_matrix with the buckets
+    :param edges: The method to generate the graph's edges:
+        - linear: linear and fast, but may not fit some of the CD algorithms, or all to all within a bucket
+        - quadratic: slow, but surely fits all the algorithms.
```

Comment:
do you think we should check `edges` param in code as it is in doc? Currently code accepts `linear` or `1` as linear and everything else as quadratic